src/main/scala/com/databricks/spark/redshift/AWSCredentials.scala

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2015 Databricks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.databricks.spark.redshift
+
+import java.net.URI
+
+import com.amazonaws.auth.InstanceProfileCredentialsProvider
+import org.apache.hadoop.conf.Configuration
+
+/**
+ * Helper class for working with AWS Credentials.
+ */
+private[redshift] case class AWSCredentials(
+    accessKey: String,
+    secretAccessKey: String,
+    sessionToken: Option[String]) {
+
+  /**
+   * Generates a credentials string for Redshift.
+   */
+  def credentialsString: String = {
+    val credentials = s"aws_access_key_id=$accessKey;aws_secret_access_key=$secretAccessKey"
+    credentials + sessionToken.map { token => s";token=$token" }.getOrElse("")
+  }
+
+}
+
+object AWSCredentials {
+
+  def load(tempPath: String, hadoopConfiguration: Configuration): AWSCredentials = {
```

Comment:
This is the trickiest part of this patch and thus requires a bit of explanation. The idea here is that the Spark <-> S3 credentials have already been configured via the URI, `hadoopConf`, or environment, so we'll attempt to read those credentials so that we can later pass them on to Redshift.

The code below tries to match the logic that the Hadoop S3 FileSystems use while resolving credentials.